src/types/pages.ts

```diff
@@ -1,52 +1,28 @@
 import { Result } from "neverthrow"
 
-import { Brand, ToBrand } from "./util"
+import { Brand, FileNameBrand, ToBrand } from "./util"
 
-type NameInfo = {
-  name: string
-  kind: string
-}
-
-type PageBrand<T extends NameInfo> = ToBrand<T, "name">
-
-export type ResourceRoomName = {
-  name: string & { __kind: "ResourceRoomName" }
-  kind: "ResourceRoomName"
-}
+export type ResourceRoomName = FileNameBrand<"ResourceRoomName">
 
-export type SubcollectionPageName = {
-  name: string & { __kind: "SubcollectionPage" }
+export type SubcollectionPageName = FileNameBrand<"SubcollectionPage"> & {
   collection: string
   subcollection: string
   kind: "SubcollectionPage"
 }
 
-export type CollectionPageName = {
-  name: string & { __kind: "CollectionPage" }
+export type CollectionPageName = FileNameBrand<"CollectionPage"> & {
   collection: string
-  kind: "CollectionPage"
```

Comment:
sanity check: Is this removal of `kind` intended?

Reply:
ye shifted to util